src/parser/on_demand/parse_to_elements.ts

```diff
@@ -44,23 +45,13 @@ export type BSONElement = [
   length: number
 ];
 
-/**
- * @experimental
- * @public
- *
- * Parses a int32 little-endian at offset, throws if it is negative
- */
-export function getSize(source: Uint8Array, offset: number): number {
-  if (source[offset + 3] > 127) {
-    throw new BSONOffsetError('BSON size cannot be negative', offset);
+const getSize = (source: Uint8Array, offset: number) => {
+  try {
```

Comment:
Why did we move `getSize` into NumberUtils?  Also, this error handling is partially redundant.  We already throw almost the same error in the `getSize` that is in number utils.

Reply:
It is a number-related utility I don't agree with our previous decision to keep it separate since it is one of the foundational number parsing operations we need in this library. The error handling keeps this API throwing a consistent error type and an error that is a subclass of BSON while keeping the getSize helper BSON agnostic.

Reply:
From slack, improved the naming to make this utility more generic to numeric parsing.